### PR TITLE
Add Memgraph as knowledge graph

### PR DESCRIPTION
**Pull request by antejavor**
### Description

Please briefly explain the changes you made.

### Pull request type

Please check what kind of PR this is:

- [ ] Fix or improvement of an existing page
- [ ] New documentation page, release related

### Related PRs and issues

PR this doc page is related to: 
(especially necessary if the PR is related to a release)

Closes:
(paste the link to the issue it closes)

### Checklist:

- [ ] Check all content with Grammarly
- [ ] Perform a self-review of my code
- [ ] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [ ] The build passes locally
- [ ] My changes generate no new warnings or errors
- [ ] Add a corresponding label
- [ ] If release-related, add a product and version label
- [ ] If release-related, add release note on product PR

### CODE REVIEW

File: pages/data-migration/knowledge-graph.mdx [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+
+import { Callout } from 'nextra/components'
+import { Card, Cards } from 'nextra/components'
+
+# Knowledge Graph
+
+From the bussines perspective, Knowledge graph represents the way of organizing
+and representing your data so it encodes the knowledge about your use-case, that
+can be explored and feched. Knowledge is encoded in the structure of the graph
+in form of **nodes** and **relationships**.
```
Comment (katarinasupe) [inline pages/data-migration/knowledge-graph.mdx:14]: and properties?

File: pages/data-migration/knowledge-graph.mdx [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+
+
+This guide will help you understand how to create a Knowledge graph using
+Memgraph.
+
+## Memgraph as a Knowledge Graph
+
+CONTEXT: Why Memgraph is a good choice for creating a Knowledge graph?
+
+Memgraph is a database based on the label property graph (LPG), this makes it a
```
Comment (katarinasupe) [inline pages/data-migration/knowledge-graph.mdx:43]: I think this is a good place to do a small RDF vs LPG section. Resources:
- https://www.notion.so/memgraph/LPG-vs-RDF-research-9d68958fdf304d2b98fd0f67cb4758dc
- https://www.figma.com/board/WGXe5aVxGaZS6huCVqZJAu/DX-2024?node-id=177-262&t=Nd6hLsgzUusxfc8c-4

File: pages/data-migration/knowledge-graph.mdx [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+flexibiles ways you can describe your domain knowledge into the graph format.
+
+This shares the same basis with the general concepts of how the graph should be
+modeled and how the data should be stored. For more details of possibiliites of
+LPG modeling, you can check the [Graph
+Modeling](/fundamentals/graph-modeling.md) section.
+
+On top of being LPG database, Memgraph is also a high-performance, in-memory
+graph database that can handle complex queries and large datasets. AS hops are
+critical part of relevance expansion in the graph, Memgraph is optimized for
```
Comment (katarinasupe) [inline pages/data-migration/knowledge-graph.mdx:54]: Relevance expansion is unknown term here. Introduce it first or don't use it in this way, make it more high-level.

File: pages/data-migration/knowledge-graph.mdx [hunk 4]
```diff
@@ -0,0 +1,571 @@
...
+graph applications.
+
+## Knowledge Graph example 
+
+For the purpose of this guide, we will use a simple example of a knowledge graph
+that represents a project management scenario. Let's  assume you have a dataset
+that contains information about people, their skills, companies they work for,
+tasks they need to do etc. Initally, this is our graph model: 
+
+- Person
```
Comment (katarinasupe) [inline pages/data-migration/knowledge-graph.mdx:65]: Not the most ideal format. Maybe put it in a code block?

File: pages/data-migration/knowledge-graph.mdx [hunk 5]
```diff
@@ -0,0 +1,571 @@
...
+    - Domain
+    - Budget
+
+- (Project)-[REQUESTED]->(Client)
+
+- Client 
+    - Name 
+    - Size
+
+In this project management scenario, we have people with different skills, cost,
```
Comment (katarinasupe) [inline pages/data-migration/knowledge-graph.mdx:97]: Repeating what the dataset is about.

File: pages/data-migration/knowledge-graph.mdx [hunk 6]
```diff
@@ -0,0 +1,571 @@
...
+and domain preference working on tasks that require specific skills, and has
+duration in man-day format. Task are part of the project, and project is
+requested by the client.
+
+Depending on the quesitons you want to ask your Knowledge graph, it can help you
```
Comment (katarinasupe) [inline pages/data-migration/knowledge-graph.mdx:102]: Ask your KG is a weird construct here. Maybe "answers you want to get from your KG" or knowledge you want to retrieve, etc..

File: pages/data-migration/knowledge-graph.mdx [hunk 7]
```diff
@@ -0,0 +1,571 @@
...
+so it was easier to infere knowledge from the graph mode.
+
+
+Imagine that you have decided to enable software agent in form of LLM to answer
+the same question. 
+
+This means that LLM needs to know how to write a query that will match that
+specific graph ontology. The second option would be to encode the knowledge into
+the graph so that LLM only needs to get the anchor place in the graph, do a
+relevance expansion and the question it needs to answer.
```
Comment (katarinasupe) [inline pages/data-migration/knowledge-graph.mdx:157]: again relevance expansion as a currently unknown term

File: pages/data-migration/knowledge-graph.mdx [hunk 8]
```diff
@@ -0,0 +1,571 @@
...
+<Cards>
+  <Card
+    title="Book a call"
+    href="https://memgraph.com/office-hours"
+  />
+</Cards>
+
+
+
+## Dataset for seeding the example
```
Comment (katarinasupe) [inline pages/data-migration/knowledge-graph.mdx:430]: Make this collapsable and place it somewhere above.

File: pages/data-migration/knowledge-graph.mdx [hunk 9]
```diff
@@ -0,0 +1,571 @@
...
+requirement for the project and a direct need for the "Data Cleaning" task. This
+indicates that her skills and interests align with her assigned
+responsibilities. 
+
+This structure suggests that team members like Jane are indeed
+working on tasks that match both their skills and their domain preferences,
+which can enhance motivation and project outcomes.
+```
+
+## Memgraph's office hours
```
Comment (katarinasupe) [inline pages/data-migration/knowledge-graph.mdx:410]: This should be last section

**Comment by katarinasupe** [comment]
Placement suggestion: Add Data modeling above Data migration and put Graph modeling and Knowledge graph under that? (+ we expand it with best practices provided by Toni J.) 

cc @antejavor

File: pages/data-migration/knowledge-graph.mdx [hunk 10]
```diff
@@ -0,0 +1,571 @@
+---
+title: Memgraph as a Knoowledge Graph
+description: Learn how to create a Knowledge Graph using Memgraph.
```
[the remark this answers is not in the document]
Comment (antejavor) [reply pages/data-migration/knowledge-graph.mdx:3]: I will leave just the title.

File: pages/data-migration/knowledge-graph.mdx (hunk 2 above)
Comment (antejavor) [reply pages/data-migration/knowledge-graph.mdx:43]: Added a sentence.

File: pages/data-migration/knowledge-graph.mdx (hunk 3 above)
Comment (antejavor) [reply pages/data-migration/knowledge-graph.mdx:54]: Rewrote that bit.

File: pages/data-migration/knowledge-graph.mdx (hunk 4 above)
Comment (antejavor) [reply pages/data-migration/knowledge-graph.mdx:65]: These will be images.

File: pages/data-migration/knowledge-graph.mdx (hunk 8 above)
Comment (antejavor) [reply pages/data-migration/knowledge-graph.mdx:430]: I am not sure how to collapse this; I didn't find any solution on the web; what do you say we upload this to the best practices repo under the kg folder?

File: pages/data-migration/knowledge-graph.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (katarinasupe) [reply pages/data-migration/knowledge-graph.mdx]: We don't use title case in any title, so I will edit this.

**Review by antejavor — COMMENTED** [review]
Much better flow 🚀 🦾

**Pull request merged**